sessiontxn/isolation/base.go

```diff
@@ -117,6 +118,25 @@ func (p *baseTxnContextProvider) GetTxnInfoSchema() infoschema.InfoSchema {
 	return p.infoSchema
 }
 
+func (p *baseTxnContextProvider) GetTxnScope() string {
+	return p.sctx.GetSessionVars().TxnCtx.TxnScope
+}
+
+func (p *baseTxnContextProvider) GetReadReplicaScope() string {
+	if txnScope := p.GetTxnScope(); txnScope != kv.GlobalTxnScope && txnScope != "" {
+		// In local txn, we should use txnScope as the readReplicaScope
+		return txnScope
+	}
+
+	if p.sctx.GetSessionVars().GetReplicaRead().IsClosestRead() {
+		// If closest read is set, we should use the scope where instance located.
+		return config.GetTxnScopeFromConfig()
+	}
```

Comment:
Should this have a higher priority than `local txn check`?

Reply:
I think they are both equal because when local txn is enabled, the "closest" replica is the "local" replica. And local txn check is a hard constraint that seems to be have "higher" priority.